src/Estimators/include/BipedalLocomotion/FloatingBaseEstimators/FloatingBaseEstimator.h

```diff
@@ -172,6 +187,26 @@ class FloatingBaseEstimator : public BipedalLocomotion::System::Advanceable<Floa
     */
     virtual bool advance() final;
 
+    /**
+     * Reset the internal state of the estimator
+     * @param[in] newState Internal state of the estimator
+     * @return True in case of success, false otherwise.
+     *
+     * @note reset and advance estimator to get updated estimator output
+     */
+    virtual bool resetEstimator(const FloatingBaseEstimators::InternalState& newState) final;
+
+    /**
+     * Reset the base pose estimate and consequently the internal state of the estimator
+     * @param[in] newBaseOrientation base link orientation as a Eigen quaternion
+     * @param[in] newBasePosition base link position
+     * @return True in case of success, false otherwise.
+     *
+     * * @note reset and advance estimator to get updated estimator output
+     */
+    virtual bool resetEstimator(const Eigen::Quaterniond& newBaseOrientation,
+                                const Eigen::Vector3d& newBasePosition) final;
```

Comment:
here you can use:
```suggestion
    virtual bool resetEstimator(Eigen::Ref<const Eigen::Quaterniond> newBaseOrientation,
                                Eigen::Ref<const Eigen::Vector3d> newBasePosition) final;
```

Reply:
`Eigen::Ref` doesn't seem to work on Quaternions.